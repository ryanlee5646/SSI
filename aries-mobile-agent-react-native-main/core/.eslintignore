node_modules
__tests__
ios
android

# generated by bob
lib/
bifold-app/
generate-app-package-json.js