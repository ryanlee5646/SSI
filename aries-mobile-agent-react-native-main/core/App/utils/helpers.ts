import {
  ConnectionRecord,
  CredentialMetadataKeys,
  CredentialRecord,
  ProofRecord,
  RequestedAttribute,
} from '@aries-framework/core'
import { useConnectionById, useCredentialById, useProofById } from '@aries-framework/react-hooks'
import { parseUrl } from 'query-string'

import { Attribute } from '../types/record'

export function parseSchema(schemaId?: string): { name: string; version: string } {
  let name = 'Credential'
  let version = ''
  if (schemaId) {
    const schemaIdRegex = /(.*?):([0-9]):([a-zA-Z .\-_0-9]+):([a-z0-9._-]+)$/
    const schemaIdParts = schemaId.match(schemaIdRegex)
    if (schemaIdParts?.length === 5) {
      name = `${schemaIdParts?.[3].replace(/_|-/g, ' ')}`
        .split(' ')
        .map((schemaIdPart) => schemaIdPart.charAt(0).toUpperCase() + schemaIdPart.substring(1))
        .join(' ')
      version = schemaIdParts?.[4]
    }
  }
  return { name, version }
}

export function credentialSchema(credential: CredentialRecord): string | undefined {
  return credential.metadata.get(CredentialMetadataKeys.IndyCredential)?.schemaId
}

export function parsedSchema(credential: CredentialRecord): { name: string; version: string } {
  return parseSchema(credentialSchema(credential))
}

/**
 * Adapted from https://stackoverflow.com/questions/3426404/create-a-hexadecimal-colour-based-on-a-string-with-javascript
 */
export function hashCode(s: string): number {
  return s.split('').reduce((hash, char) => char.charCodeAt(0) + ((hash << 5) - hash), 0)
}

export function hashToRGBA(i: number) {
  const colour = (i & 0x00ffffff).toString(16).toUpperCase()
  return '#' + '00000'.substring(0, 6 - colour.length) + colour
}

export function credentialRecordFromId(credentialId?: string): CredentialRecord | void {
  if (credentialId) {
    return useCredentialById(credentialId)
  }
}

export function connectionRecordFromId(connectionId?: string): ConnectionRecord | void {
  if (connectionId) {
    return useConnectionById(connectionId)
  }
}

export function proofRecordFromId(proofId?: string): ProofRecord | void {
  if (proofId) {
    return useProofById(proofId)
  }
}

export function getConnectionName(connection: ConnectionRecord | void): string | void {
  if (!connection) {
    return
  }
  return connection?.alias || connection?.invitation?.label
}

export function firstAttributeCredential(attributes: RequestedAttribute[], revoked = true): RequestedAttribute | null {
  if (!attributes.length) {
    return null
  }

  let first = null
  const firstNonRevoked = attributes.filter((attribute) => !attribute.revoked)[0]
  if (firstNonRevoked) {
    first = firstNonRevoked
  } else if (attributes.length && revoked) {
    first = attributes[0]
  }

  if (!first?.credentialInfo) {
    return null
  }

  return first
}

export const isRedirection = (url: string): boolean => {
  const queryParams = parseUrl(url).query
  return !(queryParams['c_i'] || queryParams['d_m'])
}

export const processProofAttributes = (
  proof: ProofRecord,
  attributes: Record<string, RequestedAttribute[]> = {}
): Attribute[] => {
  const processedAttributes = [] as Attribute[]
  const { requestedAttributes: requestedProofAttributes } = proof.requestMessage?.indyProofRequest || {}

  requestedProofAttributes?.forEach(({ name, names }, attributeName) => {
    const firstCredential = firstAttributeCredential(attributes[attributeName] || [])
    const credentialAttributes = names?.length ? names : [name || attributeName]
    credentialAttributes.forEach((attribute) => {
      processedAttributes.push({
        name: attribute,
        value: firstCredential?.credentialInfo?.attributes[attribute] || null,
        revoked: firstCredential?.revoked || false,
        credentialId: firstCredential?.credentialId,
      })
    })
  })
  return processedAttributes
}
