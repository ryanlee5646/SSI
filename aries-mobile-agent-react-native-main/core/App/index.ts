/* eslint-disable import/no-cycle */
import type { OnboardingStyleSheet } from './screens/Onboarding'

import { Agent } from '@aries-framework/core'
import AgentProvider from '@aries-framework/react-hooks'

import indyLedgers from '../configs/ledgers/indy'

import Button, { ButtonType } from './components/buttons/Button'
import CheckBoxRow from './components/inputs/CheckBoxRow'
import ErrorModal from './components/modals/ErrorModal'
import InfoTextBox from './components/texts/InfoTextBox'
import toastConfig from './components/toast/ToastConfig'
import RootStack from './navigators/RootStack'
import OnboardingPages from './screens/OnboardingPages'
import Splash from './screens/Splash'
import Terms from './screens/Terms'

export { LocalStorageKeys } from './constants'
export { initLanguages, initStoredLanguage, translationResources } from './localization'
export { ConfigurationProvider, useConfiguration } from './contexts/configuration'
export { StoreProvider, StoreContext } from './contexts/store'
export { DispatchAction } from './contexts/reducers/store'
export { ThemeProvider, useTheme } from './contexts/theme'
export { ColorPallet } from './theme'
export { theme } from './theme'
export { testIdWithKey } from './utils/testable'
export { Screens } from './types/navigators'
export { createStyles } from './screens/OnboardingPages'

export type { StoreProviderProps } from './contexts/store'
export type { Theme } from './theme'
export type { ConfigurationContext } from './contexts/configuration'
export type { Onboarding as OnboardingState } from './types/state'
export type { GenericFn } from './types/fn'
export type { AuthenticateStackParams } from './types/navigators'
export type { OnboardingStyleSheet }

export {
  indyLedgers,
  Agent,
  AgentProvider,
  Button,
  ButtonType,
  CheckBoxRow,
  InfoTextBox,
  ErrorModal,
  toastConfig,
  RootStack,
  OnboardingPages,
  Splash,
  Terms,
}
