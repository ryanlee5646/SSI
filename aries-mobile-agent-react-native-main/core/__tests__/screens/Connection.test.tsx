import { useNavigation } from '@react-navigation/core'
import { render, waitFor, fireEvent } from '@testing-library/react-native'
import React from 'react'

import ConnectionModal from '../../App/screens/Connection'
import { testIdWithKey } from '../../App/utils/testable'

jest.useFakeTimers('legacy')
jest.spyOn(global, 'setTimeout')
// jest.mock('NativeAnimatedHelp')
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper')

const props = { params: { connectionId: '123' } }

describe('ConnectionModal Component', () => {
  test('Renders correctly', async () => {
    const tree = render(<ConnectionModal route={props as any} navigation={useNavigation()} />)

    const backHomeBtn = tree.queryByTestId(testIdWithKey('BackToHome'))

    expect(tree).toMatchSnapshot()
    expect(backHomeBtn).toBeNull()
  })

  test('Updates after delay', async () => {
    const tree = render(<ConnectionModal route={props as any} navigation={useNavigation()} />)

    await waitFor(() => {
      jest.runAllTimers()
    })

    expect(tree).toMatchSnapshot()
  })

  test('Dismiss on demand', async () => {
    const tree = render(<ConnectionModal route={props as any} navigation={useNavigation()} />)

    await waitFor(() => {
      jest.runAllTimers()
    })

    const backHomeBtn = tree.getByTestId(testIdWithKey('BackToHome'))
    fireEvent(backHomeBtn, 'press')

    expect(tree).toMatchSnapshot()
  })
})
