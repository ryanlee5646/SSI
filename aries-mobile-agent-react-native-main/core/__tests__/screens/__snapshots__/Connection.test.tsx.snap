// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionModal Component Dismiss on demand 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  transparent={true}
  visible={false}
>
  <RNCSafeAreaView
    style={
      Array [
        Object {
          "backgroundColor": "#000000",
          "flexDirection": "column",
          "flexGrow": 1,
          "paddingHorizontal": 25,
          "paddingTop": 20,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 26,
              "fontWeight": "bold",
            },
            Object {
              "fontWeight": "normal",
              "marginTop": 90,
              "textAlign": "center",
            },
          ]
        }
        testID="com.ariesbifold:id/CredentialOnTheWay"
      >
        Connection.JustAMoment
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "marginTop": 20,
          },
          Object {
            "alignItems": "center",
            "justifyContent": "flex-end",
            "minHeight": 250,
          },
        ]
      }
    >
      <View>
        <
          fill="#FFFFFF"
          height={250}
          width={250}
        />
        <View
          style={
            Object {
              "position": "absolute",
              "transform": Array [
                Object {
                  "rotate": "360deg",
                },
              ],
            }
          }
        >
          <
            fill="#FFFFFF"
            height={250}
            width={250}
          />
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</Modal>
`;

exports[`ConnectionModal Component Renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <RNCSafeAreaView
    style={
      Array [
        Object {
          "backgroundColor": "#000000",
          "flexDirection": "column",
          "flexGrow": 1,
          "paddingHorizontal": 25,
          "paddingTop": 20,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 26,
              "fontWeight": "bold",
            },
            Object {
              "fontWeight": "normal",
              "marginTop": 90,
              "textAlign": "center",
            },
          ]
        }
        testID="com.ariesbifold:id/CredentialOnTheWay"
      >
        Connection.JustAMoment
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "marginTop": 20,
          },
          Object {
            "alignItems": "center",
            "justifyContent": "flex-end",
            "minHeight": 250,
          },
        ]
      }
    >
      <View>
        <
          fill="#FFFFFF"
          height={250}
          width={250}
        />
        <View
          style={
            Object {
              "position": "absolute",
              "transform": Array [
                Object {
                  "rotate": "0deg",
                },
              ],
            }
          }
        >
          <
            fill="#FFFFFF"
            height={250}
            width={250}
          />
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</Modal>
`;

exports[`ConnectionModal Component Updates after delay 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <RNCSafeAreaView
    style={
      Array [
        Object {
          "backgroundColor": "#000000",
          "flexDirection": "column",
          "flexGrow": 1,
          "paddingHorizontal": 25,
          "paddingTop": 20,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 26,
              "fontWeight": "bold",
            },
            Object {
              "fontWeight": "normal",
              "marginTop": 90,
              "textAlign": "center",
            },
          ]
        }
        testID="com.ariesbifold:id/CredentialOnTheWay"
      >
        Connection.JustAMoment
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "marginTop": 20,
          },
          Object {
            "alignItems": "center",
            "justifyContent": "flex-end",
            "minHeight": 250,
          },
        ]
      }
    >
      <View>
        <
          fill="#FFFFFF"
          height={250}
          width={250}
        />
        <View
          style={
            Object {
              "position": "absolute",
              "transform": Array [
                Object {
                  "rotate": "360deg",
                },
              ],
            }
          }
        >
          <
            fill="#FFFFFF"
            height={250}
            width={250}
          />
        </View>
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "normal",
          },
          Object {
            "marginTop": 20,
            "textAlign": "center",
          },
        ]
      }
      testID="com.ariesbifold:id/TakingTooLong"
    >
      Connection.TakingTooLong
    </Text>
    <View
      style={
        Array [
          Object {
            "marginBottom": 20,
            "marginTop": "auto",
          },
        ]
      }
    >
      <View
        accessibilityLabel="Loading.BackToHome"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "#42803E",
            "borderRadius": 4,
            "borderWidth": 2,
            "opacity": 1,
            "padding": 16,
          }
        }
        testID="com.ariesbifold:id/BackToHome"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#42803E",
                "fontSize": 18,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              false,
            ]
          }
        >
          Loading.BackToHome
        </Text>
      </View>
    </View>
  </RNCSafeAreaView>
</Modal>
`;
